sdk-libs/halo-notifications/src/main/java/com/mobgen/halo/android/notifications/services/NotificationService.java

```diff
@@ -183,6 +188,15 @@ private HaloNotificationDecorator createNotificationDecorator() {
     }
 
     /**
+     * Set the notification id generator.
+     *
+     * @param idGenerator The id generator.
+     */
+    public static void setIdGenerator(@NonNull NotificationIdGenerator idGenerator) {
+        mIdGenerator = idGenerator;
```

Comment:
Here is where you should test for non null param not in the API. This way you avoid someone to call this on their side